src/analysis/pump_bep.jl

```diff
@@ -0,0 +1,59 @@
+"""
+Calculate the best-efficiency point (BEP) for the pumps in the water network. The BEP flow,
+head, and power are added to the pumps' dictionaries.
+"""
+function calc_pump_bep!(data::Dict{String,<:Any})
+    for (pump_id, pump) in data["pump"]
+        q_bep = NaN
+
+        # Calculate best efficiency point efficiencies.
+        if haskey(pump, "efficiency_curve")
+            eff_curve_tuples = pump["efficiency_curve"]
+            eff_curve = array_from_tuples(eff_curve_tuples)
+            
+            # fit efficiency curve to determine q_bep and eta_bep
+            A = hcat(eff_curve[:, 1].^2, eff_curve[:, 1])
+            a, b = A \ eff_curve[:, 2]
+            q_bep, eta_bep = -0.5*b * inv(a), -0.25*b^2 * inv(a)
+        else
+            # presume the single-value for efficiency is at the BEP
+            eta_bep = pump["efficiency"]
+        end
+
+        # Calculate best efficiency point head gains.
+        head_curve_tuples = pump["head_curve"]
+        head_curve = array_from_tuples(head_curve_tuples)
+
+        if isnan(q_bep)
+            # an efficiency curve was not provided, and so we must determine q_bep from the
```

Comment:
Shall we consider situations where you don't have head curve either and only have a single-point power rating? I do think it's reasonable to assume that a head curve is provided here though.